src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -362,6 +362,12 @@
     @QueryParam("geoidElevation")
     private Boolean geoidElevation;
 
+    /**
+     * If request date is invalid, apply the provided strategy to come up with a valid date.
+     */
+    @QueryParam("invalidDateStrategy")
+    protected String invalidDateStrategy;
+
```

Comment:
I don´t like the idea of injecting a date parsing strategy. The OTP is a backend server and the clients should send well formatted dates - if not the OTP should fail with BAD_REQUEST. Dates that are syntactically correct formated, but not *real* dates,  like the "2018-02-29" and "2018-01-99", may be accepted - and converted to the correct date. I don´t think we need a strategy to do this - the `DateUtils.toDate()` could be changed to do this. 

I no date and/or time is passed in, then it is OK to use the current date and/or time as default. 
  
The `DateUtils.toDate` is unfortunately error prune. The "10:30 AM:34" is a valid time and the "10:30:34 AM" is not.